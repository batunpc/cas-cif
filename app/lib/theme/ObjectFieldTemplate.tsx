import Grid from "@button-inc/bcgov-theme/Grid";
import FormBorder from "./components/FormBorder";

const DefaultDescriptionField = ({ id, description }) => (
  <span id={id}>{description}</span>
);

const ObjectFieldTemplate = (props) => {
  const DescriptionField = props.DescriptionField || DefaultDescriptionField;

  return (
    <Grid cols={10}>
      <Grid.Row>
        <Grid.Col span={10} id="test">
          <FormBorder
            title={props.title ? props.title : props.uiSchema["ui:title"]}
          >
            {props.description && (
              <DescriptionField
                id={`${props.idSchema.$id}__description`}
                description={props.description}
                formContext={props.formContext}
              />
            )}
            {props.properties.map((prop) => prop.content)}
          </FormBorder>
        </Grid.Col>
      </Grid.Row>
    </Grid>
  );
};

export default ObjectFieldTemplate;
