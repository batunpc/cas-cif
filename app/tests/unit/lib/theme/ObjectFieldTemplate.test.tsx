import { render } from "@testing-library/react";
import ObjectFieldTemplate from "lib/theme/ObjectFieldTemplate";
import React from "react";

describe("The Object Field Template", () => {
  it("renders a default TitleField and DescriptionField", () => {
    const props = {
      idSchema: { $id: "test-id" },
      uiSchema: {},
      properties: [],
      type: "object",
      title: "Test Title",
      description: "Test Description",
    };

    const componentUnderTest = render(<ObjectFieldTemplate {...props} />);

    expect(componentUnderTest.container).toMatchSnapshot();
  });

  it("renders the provided TitleField and DescriptionField with the provided title and description", () => {
    const props = {
      idSchema: { $id: "test-id" },
      uiSchema: {},
      properties: [],
      type: "object",
      title: "Second title",
      description: "Another Description",
      TitleField: ({ title }) => <p>Test {title}</p>,
      DescriptionField: ({ description }) => <p>Test {description}</p>,
    };

    const componentUnderTest = render(<ObjectFieldTemplate {...props} />);
    expect(componentUnderTest.container).toMatchSnapshot();
  });
});
