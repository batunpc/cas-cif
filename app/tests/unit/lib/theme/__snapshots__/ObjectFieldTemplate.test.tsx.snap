// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Object Field Template renders a default TitleField and DescriptionField 1`] = `
<div>
<<<<<<< HEAD
  <fieldset
    class="sc-dkPtRN bfJYwN pg-fieldset"
    id="test-id"
=======
  <div
    class="sc-jSgvzq dUsOVZ pg-grid-container"
    cols="10"
>>>>>>> c1952b7 (test: snapshots)
  >
    <div
      class="sc-gKseQn sc-iBPTik diOpKM cirWLn pg-grid-row"
    >
      <div
        class="sc-fubCzh sc-pGacB cMGSGp elfPxV pg-grid-col"
        id="test"
        width="100"
      >
        <fieldset
          class="jsx-3408663875"
          id="formborder"
        >
          <legend
            class="jsx-3408663875"
          >
            Test Title
          </legend>
          <span
            id="test-id__description"
          >
            Test Description
          </span>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Object Field Template renders the provided TitleField and DescriptionField with the provided title and description 1`] = `
<div>
<<<<<<< HEAD
  <fieldset
    class="sc-dkPtRN bfJYwN pg-fieldset"
    id="test-id"
=======
  <div
    class="sc-jSgvzq dUsOVZ pg-grid-container"
    cols="10"
>>>>>>> c1952b7 (test: snapshots)
  >
    <div
      class="sc-gKseQn sc-iBPTik diOpKM cirWLn pg-grid-row"
    >
      <div
        class="sc-fubCzh sc-pGacB cMGSGp elfPxV pg-grid-col"
        id="test"
        width="100"
      >
        <fieldset
          class="jsx-3408663875"
          id="formborder"
        >
          <legend
            class="jsx-3408663875"
          >
            Second title
          </legend>
          <p>
            Test 
            Another Description
          </p>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
